android/scripts/make_rust.sh

```diff
@@ -1,52 +1,91 @@
 #!/bin/bash
 
 # Run from repo's root folder (relative to Gradle build file. TODO do this in Gradle)
-cd ../..
-echo "Running make_rust.sh from $(pwd)"
+
+if [ -d "android/core/gradle" ]; then
+    echo "Starting..."
+else
+    echo "Error: this script must be run in project root"
+    exit 5
+fi
+
+root=$(pwd)
+
+echo "Running make_rust.sh from $root"
 
 # Building  ###########################################################
 
 # See https://github.com/Co-Epi/app-backend-rust/wiki/Building-library-for-Android
 
+if test -z "$PATH_TO_ANDROID_REPO"; then
+    echo "Environment variable PATH_TO_ANDROID_REPO not set. Please set the variable using: 'export PATH_TO_ANDROID_REPO=<your_path_here>'"
+    exit 6
+else
+    echo "\$PATH_TO_ANDROID_REPO is $PATH_TO_ANDROID_REPO"
+    if [ -d "$PATH_TO_ANDROID_REPO" ]; then
+        echo "Will copy files to ${PATH_TO_ANDROID_REPO}..."
+    else
+        echo "Error: ${PATH_TO_ANDROID_REPO} not found. Make sure that you have checked out the corresponding Android project and that the path specified in PATH_TO_ANDROID_REPO environment variable is correct"
+        exit 7
+    fi
+fi
+
+# Order of target_triples has to match the order of arhitectures:
+# aarch64-linux-android --> arm64-v8a
+# armv7-linux-androideabi --> armeabi
+# x86_64-linux-android --> x86_64
+# i686-linux-android --> x86
+
+target_triples=(aarch64-linux-android armv7-linux-androideabi x86_64-linux-android i686-linux-android)
+architectures=(arm64-v8a armeabi x86_64 x86)
+
+if [ ${#target_triples[@]} != ${#architectures[@]} ]; then
```

Comment:
Workaround for a dictionary, which kind of proves that shell is the wrong tool for this.

Reply:
It is a workaround for dictionary. Dictionaries are supported in Bash 4, but macs only come with Bash 3 (due to licensing issues).
I still need to research alternatives (cargo-make), until then I would like to merge the current PR.

Reply:
Follow-up of comments below: shouldn't be merged, since it breaks the CI and the releases, and it's not blocking anything.

Reply:
Local seems fine with latest fixes. Not sure how to go about testing CI.